specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-07-01-preview/common/EntityTypes.json

```diff
@@ -1000,6 +1010,96 @@
           "readOnly": true,
           "type": "array",
           "x-ms-identifiers": []
+        },
+        "owners": {
+          "description": "A list of owners of the IoTDevice entity.",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true,
+          "type": "array",
+          "x-ms-identifiers": []
+        },
+        "nicEntityIds": {
+          "description": "A list of Nic entity ids of the IoTDevice entity.",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true,
+          "type": "array",
+          "x-ms-identifiers": []
+        },
+        "site": {
+          "description": "The site of the device",
+          "readOnly": true,
+          "type": "string"
+        },
+        "zone": {
+          "description": "The zone location of the device within a site",
+          "readOnly": true,
+          "type": "string"
+        },
+        "sensor": {
+          "description": "The sensor the device is monitored by",
+          "readOnly": true,
+          "type": "string"
+        },
+        "deviceSubType": {
+          "description": "The subType of the device ('PLC', 'HMI', 'EWS', etc.)",
+          "readOnly": true,
+          "type": "string"
+        },
+        "importance": {
+          "description": "Device importance, determines if the device classified as 'crown jewel'",
+          "enum": [
+            "Unknown",
+            "Low",
+            "Normal",
+            "High"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "DeviceImportance",
+            "values": [
+              {
+                "description": "Unknown - Default value",
+                "value": "Unknown"
+              },
+              {
+                "description": "Low",
+                "value": "Low"
+              },
+              {
+                "description": "Normal",
+                "value": "Normal"
+              },
+              {
+                "description": "High",
+                "value": "High"
+              }
+            ]
+          }
+        },
+        "purdueLayer": {
+          "description": "The Purdue Layer of the device",
+          "readOnly": true,
+          "type": "string"
+        },
+        "isAuthorized": {
+          "description": "Determines whether the device classified as authorized device",
+          "readOnly": true,
+          "type": "boolean"
+        },
+        "isProgramming": {
+          "description": "Determines whether the device classified as programming device",
+          "readOnly": true,
+          "type": "boolean"
+        },
+        "isScanner": {
+          "description": "Is the device classified as a scanner device",
+          "readOnly": true,
```

Comment:
consider upgrading to enum as explained here : [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)


Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].